Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/DocumentObj/SetIshDocumentObj.Tests.ps1

```diff
@@ -0,0 +1,759 @@
+			
+			$ishRequiredCurrentMetadata = Set-IshRequiredCurrentMetadataField -IshSession $ishSession -Name "FSTATUS" -Level Lng -ValueType Element -Value $ishStatusDraft
+			$ishObjectArray = $ishObjectAB | Set-IshDocumentObj -IshSession $ishSession -RequiredCurrentMetadata $ishRequiredCurrentMetadata
+			[xml]$xmlRevisionInfoA = $ishSession.DocumentObj25.RetrieveRevisionInfoByIshLngRefs(@($ishObjectA.LngRef), "")
+			[xml]$xmlRevisionInfoB = $ishSession.DocumentObj25.RetrieveRevisionInfoByIshLngRefs(@($ishObjectB.LngRef), "")
+
+			$ishObjectArray.Count | Should Be 2
+			$xmlRevisionInfoA.SelectNodes("//ishobjects/ishobject/ishchanges/ishchange").Count | Should Be 2
+			$xmlRevisionInfoB.SelectNodes("//ishobjects/ishobject/ishchanges/ishchange").Count | Should Be 2
+		}
+		
+		It "Resubmit blob, provide multiple objects via pipeline with RequiredCurrentMetadata rejected" {
+			$ishObjectA = Add-IshDocumentObj -IshSession $ishSession -IshFolder $ishFolderTopic -IshType ISHModule -Lng $ishLng -Metadata $ishTopicMetadata -FileContent $ditaTopicFileContent
+			$ishObjectB = Add-IshDocumentObj -IshSession $ishSession -IshFolder $ishFolderTopic -IshType ISHModule -Lng $ishLng -Metadata $ishTopicMetadata -FileContent $ditaTopicFileContent
+			$ishObjectAB = @($ishObjectA, $ishObjectB) | Get-IshDocumentObj -IshSession $ishSession -IncludeData
+			
+			$ishRequiredCurrentMetadata = Set-IshRequiredCurrentMetadataField -IshSession $ishSession -Name "FSTATUS" -Level Lng -ValueType Element -Value $ishStatusReleased
+			{$ishObjectAB | Set-IshDocumentObj -IshSession $ishSession -RequiredCurrentMetadata $ishRequiredCurrentMetadata} | 
+						Should Throw '[-106011] The supplied expected metadata value "Released" does not match the current database value "Draft" so we rolled back your operation. To make the operation work you should make sure your value matches the latest database value. [f:158 fe:FSTATUS ft:LOV] [106011;InvalidCurrentMetadata]'
+		}
+		
+		It "Resubmit blob and set Metadata, provide multiple objects via pipeline" {
+			$ishObjectA = Add-IshDocumentObj -IshSession $ishSession -IshFolder $ishFolderTopic -IshType ISHModule -Lng $ishLng -Metadata $ishTopicMetadata -FileContent $ditaTopicFileContent
+			$ishObjectB = Add-IshDocumentObj -IshSession $ishSession -IshFolder $ishFolderTopic -IshType ISHModule -Lng $ishLng -Metadata $ishTopicMetadata -FileContent $ditaTopicFileContent
+			$ishObjectAB = @($ishObjectA, $ishObjectB) | Get-IshDocumentObj -IshSession $ishSession -IncludeData
+		
+			$updatedTitle = $ishObjectA.ftitle_logical_value + "...updated"								
+			$ishMetadataFieldsSet = Set-IshMetadataField -IshSession $ishSession -Name "FTITLE" -Level "Logical" -Value $updatedTitle
+			$ishObjectArray = $ishObjectAB | Set-IshDocumentObj -IshSession $ishSession -Metadata $ishMetadataFieldsSet
+			[xml]$xmlRevisionInfoA = $ishSession.DocumentObj25.RetrieveRevisionInfoByIshLngRefs(@($ishObjectA.LngRef), "")
```

Comment:
Right, but with checking number of revisions I am testing if the blob was submitted by Set-IshDocumentObj or not. Since we have variations in DocumentObj25 api (Update, SetMetadata) I want to make sure the proper code path was used in Set-IshDocumentObj.

Reply:
Will try to simplify using  Get-IshDocumentObj